pydantic/networks.py

```diff
@@ -610,16 +599,21 @@ def host(self) -> str:
         return self._url.host  # type: ignore
 
 
-class FileUrl(_BaseUrl):
+class FileUrl(AnyUrl):
     """A type that will accept any file URL.
 
     * Host not required
     """
 
     _constraints = UrlConstraints(allowed_schemes=['file'])
 
+    @property
+    def host(self) -> str | None:  # pyright: ignore[reportIncompatibleMethodOverride]
+        """The host part of the URL, or `None`."""
+        return self._url.host
+
```

Comment:
Have to add this for the case where `host_required=Fals` because `AnyUrl` has `host` typed as `str`, so we indicate the optional nature here.